lxd/instance/instance_utils.go

```diff
@@ -872,3 +872,31 @@ func SuitableArchitectures(s *state.State, project string, req api.InstancesPost
 	// No other known types
 	return nil, fmt.Errorf("Unknown instance source type: %s", req.Source.Type)
 }
+
+// ValidName validates an instance name. There are different validation rules for instance snapshot names
+// so it takes an argument indicating whether the name is to be used for a snapshot or not.
+func ValidName(instanceName string, isSnapshot bool) error {
+	if isSnapshot {
+		parentName, snapshotName, _ := shared.InstanceGetParentAndSnapshotName(instanceName)
+		err := shared.ValidHostname(parentName)
+		if err != nil {
+			return errors.Wrap(err, "Invalid instance name")
+		}
+
+		err = shared.ValidHostname(snapshotName)
```

Comment:
@stgraber this enforces the host name validation rules on the snapshot name part of a snapshot instance name. Is this OK (are there cases where a snapshot might have names that do not comply with those rules)?

Reply:
Yeah, it's quite likely that some people have used dots for timestamps or names longer than 64 chars. There's no good reason for us to restrict snapshots the same way we restrict instances.

So a normal check of ensuring no slash or spaces should be enough.

Reply:
@stgraber thanks thats fixed now.